types/views/views.go

```diff
@@ -329,13 +329,88 @@ func SliceEqualAnyOrder[T comparable](a, b Slice[T]) bool {
 	return true
 }
 
-// MapOf returns a view over m. It is the caller's responsibility to make sure K
-// and V is immutable, if this is being used to provide a read-only view over m.
-func MapOf[K comparable, V comparable](m map[K]V) Map[K, V] {
-	return Map[K, V]{m}
+// MapSlice is a view over a map whose values are slices.
+type MapSlice[K comparable, V any] struct {
+	// ж is the underlying mutable value, named with a hard-to-type
+	// character that looks pointy like a pointer.
+	// It is named distinctively to make you think of how dangerous it is to escape
+	// to callers. You must not let callers be able to mutate it.
+	ж map[K][]V
+}
+
+// MapSliceOf returns a MapSlice for the provided map. It is the caller's
+// responsibility to make sure V is immutable.
+func MapSliceOf[K comparable, V any](m map[K][]V) MapSlice[K, V] {
+	return MapSlice[K, V]{m}
 }
 
-// Map is a view over a map whose values are immutable.
+// Has reports whether k has an entry in the map.
+func (m MapSlice[K, V]) Has(k K) bool {
```

Comment:
We use "Contains" elsewhere, no? https://pkg.go.dev/tailscale.com/util/set#Set.Contains

And ContainsFunc in this package.

Reply:
yeah looks like we haven't been consistent here. I can add a Contains and mark `Has` has deprecated.